frontend/pkg/frontend/frontend.go

```diff
@@ -237,6 +237,80 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 	writer.WriteHeader(http.StatusOK)
 }
 
+func (f *Frontend) GetNodePoolOfClusterByName(writer http.ResponseWriter, request *http.Request) {
+	ctx := request.Context()
+
+	versionedInterface, err := VersionFromContext(ctx)
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	f.logger.Info(fmt.Sprintf("%s: GetNodePools", versionedInterface))
+
+	// URL path is already lowercased by middleware.
+	resourceID := request.URL.Path
+	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
+	clusterDoc, err := f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			f.logger.Error(fmt.Sprintf("existing cluster document not found for cluster: %s on GET node pool by name", resourceID))
+			writer.WriteHeader(http.StatusNoContent)
+			return
+		} else {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+	}
```

Comment:
I think it would be worth adding the Cluster ID to the node pool document so only one CosmosDB query is required to retrieve node pool data from Cluster Service.  Would simplify the logic both here and in the DELETE handler.

Reply:
+1 to adding cluster id to node pool doc

I dont know if there is an ADR/DDR i missed but is it possibly more beneficial to leverage the cluster docs to store node pool information and forgo node pool docs altogether? Since node pools are tied to clusters, getting the cluster doc would provide us the ID and potentially all node pools tied to it? 

maybe not for MVP but a thought.

Reply:
I'm not sure I would combine multiple resources into one document, since that runs a higher risk of database commit conflicts.  But we could potentially combine the cluster and node pool database containers into a more generalized "resource" container.  The only significant schema change would be capturing how to uniquely identify any Cluster Service resource.  My suggestion there is to use their own [REST API](https://api.openshift.com/#/default/get_api_clusters_mgmt_v1_clusters__cluster_id__node_pools__node_pool_id_) as a guide.

e.g.
```
ExternalID: {azure_resource_id}
InternalID: "/clusters/{cluster_id}/node_pools/{node_pool_id}"
```

Reply:
I understand the point of trying to avoid one DB call, but I am not completely sure about that because that will imply that when we want to query a NodePool, we skip the retrieval of the ClusterDoc from Cosmos, assuming it is already there and it is okay, something I am not 100% sure we should do. I am just trying to think about that in terms of consistency, to check if there is something strange that could happen by not checking the existence of the cluster doc before the nodepool doc. 🤔

Reply:
After the conversation in the Slack Thread, I am leaving the CosmosDB access as it is, as we will need to check the Provisioning state of the cluster and check that it is different than _ProvisioningStateDeleting_ once we have this info available in database.HCPOpenShiftClusterDocument (not ready yet).